crates/core/tedge_mapper/src/sm_c8y_mapper/error.rs

```diff
@@ -1,21 +1,16 @@
+use agent_interface::error::*;
 use c8y_smartrest::error::{SmartRestDeserializerError, SmartRestSerializerError};
 
-#[derive(thiserror::Error, Debug)]
-pub enum MapperTopicError {
-    #[error("Topic {topic} is unknown.")]
-    UnknownTopic { topic: String },
-}
-
 #[derive(thiserror::Error, Debug)]
 pub enum SMCumulocityMapperError {
     #[error("Invalid MQTT Message.")]
     InvalidMqttMessage,
 
     #[error(transparent)]
-    InvalidTopicError(#[from] MapperTopicError),
+    InvalidTopicError(#[from] TopicError),
```

Comment:
I would rename the error type and make explicit that this is related to the agent (minor).

``` suggestion
    InvalidTopicError(#[from] agent_interface::UnknowTopic),
```